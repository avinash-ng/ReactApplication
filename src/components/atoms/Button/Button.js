import React from 'react'
import PropTypes from 'prop-types'
const Button = (props ) => {
    return (
        <button className="a_button" className={props.className} onClick={props.onClicked}>{props.buttonName}</button>
    )
}

Button.propTypes = {
    buttonName: PropTypes.string.isRequired
}

export default Button
