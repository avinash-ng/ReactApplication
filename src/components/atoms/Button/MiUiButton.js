import React from 'react'
import { Button } from "@material-ui/core"
const MiUiButton = props => {
    return (
        <Button className={props.className} style={props.style} variant={props.variant} onClick={props.onClick}>{props.buttonName}</Button> 
    )
}
export default MiUiButton