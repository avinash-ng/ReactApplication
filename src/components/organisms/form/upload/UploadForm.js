import React, {useState} from 'react'
import InputField from '../../../atoms/Input/Input'
import HTMLButton from '../../../atoms/Button/Button'
import axios from 'axios'
import './upload.css'
import $ from 'jquery'
import {makeStyles} from '@material-ui/core/styles'
import Button from '../../../atoms/Button/MiUiButton'
export var uploadFormSubmit = false


const useStyles = makeStyles((theme) => ({

    label: {
        width: '100px'
    },
    span: {
        backgroundColor: '#428bca',
        fontSize: '20px',
        padding: '10px',
        borderRadius: '4px',
        width: '100px',
        paddingTop: '8px',
        paddingBottom: '11px',
        color: 'white',
        cursor: 'pointer',
    },
    heading: {
        textAlign: 'center'
    }, 
    button: {
        marginLeft: 'auto',
        marginTop: '50px',
        fontSize: '20px',
        justifyContent: 'center',
        backgroundColor: 'black',
        color: 'white'
    },
    mainDiv: {
        width: '750px',
        margin: 'auto',
        alignItems: 'center',
        justifyContent: 'center'
    },
    formcontrol: {
        height: '38px',
        width: '375px'
    },
    inputgroup: {
        marginBottom: '20px',
        textAlign: 'center'
    },
    helpblock: {
        fontSize: '20px',
        marginLeft: '50px',
    }

}))
const UploadForm = () => {
    const classes = useStyles()
    const [file, setFile] = useState(null)
    const onChange = (e) => {
        setFile(e.target.files[0])
        document.getElementsByClassName("form-control")[0].value = e.target.files[0].name

    }

    const onFormSubmit = () => {
        uploadFormSubmit = true
        if (file !== null) {
            var formData = new FormData();
            formData.append('productImage', file);
            console.log(formData.productImage)
            // var data = { binary: file }
            if (file !== undefined) {
                axios.post("http://localhost:3001/image", formData)
            }
        }

    }

    const resetForm = () => {
        
    }

    return (
        <form name = "myform" onSubmit={onFormSubmit()}>
        
            <div className="col-lg-6 col-sm-6 col-12" className="upload-main-div" className={classes.mainDiv}>
                <h1 className={classes.heading}>File Upload</h1>
            <h4 className={classes.heading}> Input Groups</h4>
                    <div className="input-group" className={classes.inputgroup}>
                    
                        <label className="input-group-btn" className={classes.label} >
                    
                        <span id="span" className="btn btn-primary" className={classes.span}>
                                Browse&hellip; <input type="file" style={{ display: 'none' }} onChange={onChange}/>
                        </span>
                        </label>
                    
                        <input type="text" className="form-control"  />
                    </div>
                <span style={{ justifyContent: 'center' }} className="help-block" className={classes.span}>
                    Try uploading a file and you could see in gallery
            </span> <br/>
                
                <Button className={classes.button} type="submit" name="upload" buttonName="upload"></Button>
        </div>
        </form>
    )
}

export default UploadForm