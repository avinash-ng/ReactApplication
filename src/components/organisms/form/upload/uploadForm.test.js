const { shallow, configure, mount } = require("enzyme")
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import Button from "../../../atoms/Button/Button";
import sinon from 'sinon'
import { render, fireEvent } from "@testing-library/react";
import UploadForm, { uploadFormSubmit } from './UploadForm'
import { confirmUploadForm } from './UploadForm'
configure({ adapter: new Adapter() });
describe('testing <UploadForm /> ', () => {
    it('renders <UploadForm /> correctly', () => {
        const RenderComponent = shallow(<UploadForm> </UploadForm>)
        expect(RenderComponent.exists()).toBe(true)
    })

    it('form submitting functionality', () => {
        const deleteFunction = sinon.spy()
        const RenderComponent = mount(<UploadForm > </UploadForm>)

        RenderComponent.find('form').simulate('submit')

        expect(uploadFormSubmit).toBe(true)


    })

    it('matches snapshot', () => {
        const RenderComponent = shallow(<UploadForm></UploadForm>)
        expect(RenderComponent).toMatchSnapshot()
    })
})