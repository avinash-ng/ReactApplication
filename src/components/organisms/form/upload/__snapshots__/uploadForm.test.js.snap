// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <UploadForm />  matches snapshot 1`] = `ShallowWrapper {}`;
